Data/Vector/Primitive.hs

```diff
@@ -137,6 +137,9 @@ module Data.Vector.Primitive (
 
   -- ** Other vector types
   G.convert,
+#if __GLASGOW_HASKELL__ >= 708
+  unsafeCoerceVector,
```

Comment:
Having conditional API is likely to make lifes of downstream dependencies more difficult. It may be better to export `unsafeCoerceVector` unconditionally and define it for old GHCs without `Coercible` constraint.